src/Microsoft.Diagnostics.ExtensionCommands/Host/RuntimesCommand.cs

```diff
@@ -30,16 +30,28 @@ public class RuntimesCommand : CommandBase
         [Option(Name = "--netcore", Aliases = new string[] { "-netcore", "-c" }, Help = "Switches to the .NET Core or .NET 5+ runtime if exists.")]
         public bool NetCore { get; set; }
 
+        [Option(Name = "--all", Aliases = new string[] { "-a" }, Help = "Forces all runtimes to be enumerated.")]
+        public bool All { get; set; }
+
         public override void Invoke()
         {
             if (NetFx && NetCore)
             {
                 throw new DiagnosticsException("Cannot specify both -netfx and -netcore options");
             }
+            RuntimeEnumerationFlags flags = RuntimeEnumerationFlags.Default;
+
+            if (All)
+            {
+                // Force all runtimes to be enumerated. This requires a target flush.
```

Comment:
Why does forcing all runtimes to be enumerated require a target flush?

Reply:
The previous set of runtimes that are enumerated are cached.  Flushing the target is the only way to do that because the runtime instance could not only be cached in the RuntimeService implementation but other services/code could cache it.